### PR TITLE
feat #SB-25173: new contentAdditionMode for copyCollections

### PR DESCRIPTION
**Pull request by bharatkashyap**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-25173" title="SB-25173" target="_blank">EpicSB-25173</a>  Refactoring the project creation flow.
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Nodejs 8.11.2
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/helpers/programHelper.js [hunk 1]
```diff
@@ -630,6 +630,20 @@ class ProgramServiceHelper {
 
     const collections = _.get(data, 'config.collections');
     const collectionIds = _.map(collections, 'id');
+    if(_.includes(_.get(data, 'config.contentAdditionMode'), 'new')) {
```
Comment (pallakartheekreddy) [inline src/helpers/programHelper.js:633]: @bharatkashyap are you storing `contentAdditionMode` in program config?

**Review by pallakartheekreddy — CHANGES_REQUESTED** [review]

File: src/helpers/programHelper.js (hunk 1 above)
Comment (bharatkashyap) [reply src/helpers/programHelper.js:633]: Yes @pallakartheekreddy

Comment (bharatkashyap) [reply src/helpers/programHelper.js]: @pallakartheekreddy - Have updated this. Please review

File: src/service/programService.js [hunk 2]
```diff
@@ -406,8 +405,64 @@ function unlistPublishProgram(req, response) {
         return response.status(400).send(errorResponse(errObj,errCode+errorCodes.CODE4));
       }
     };
-
-    programServiceHelper.copyCollections(res, data.request.channel, req.headers, cb);
+      model.configuration.findOne({
```
Comment (pallakartheekreddy) [inline src/service/programService.js:408]: @bharatkashyap Can you please create a function to get a configuration that will take input as key and return value as output. So we can use that function in some other place as well.

File: src/service/programService.js [hunk 3]
```diff
@@ -406,8 +405,64 @@ function unlistPublishProgram(req, response) {
...
+        where: {
+          key: 'programTargetObjectMap',
+          status: 'active'
+        }
+      }).then(config => {
+        if(config) {
+          let targetObjectMap = JSON.parse(config.value || (config.dataValues && config.dataValues.value));
+          let targetCollectionPrimaryCategory = _.get(res, 'targetcollectionprimarycategories') || _.get(res, 'dataValues.targetcollectionprimarycategories');
+          if(!_.isEmpty(targetCollectionPrimaryCategory) || !_.isEmpty(targetObjectMap)) {
```
Comment (pallakartheekreddy) [inline src/service/programService.js:417]: @bharatkashyap please refactor this code. same write another function which will return `targetAdditionMode`

**Review by pallakartheekreddy — CHANGES_REQUESTED** [review]

File: src/service/programService.js (hunk 2 above)
Comment (bharatkashyap) [reply src/service/programService.js]: @pallakartheekreddy - Have done

File: src/service/programService.js (hunk 3 above)
Comment (bharatkashyap) [reply src/service/programService.js]: @pallakartheekreddy - Have done this

**Pull request closed without merging**